### PR TITLE
fix downloading bugs

### PR DESCRIPTION
**Pull request by murugaratham**
I'm not sure if your existing codes have any issue to download, but when i run `download package` task in vscode, it hangs, i did some basic tests and all seems to work fine now.

Your validate tarballs would also sometimes have a race condition of validation failing, and it doesn't retry

### CODE REVIEW

File: lib/crawler.js [hunk 1]
```diff
@@ -75,6 +75,7 @@ async function _retrievePackageVersion({
     version,
     outputPrefix = ''
 }) {
+    version = version || version.version;
```
Comment (Meir017) [inline lib/crawler.js]: when does `version.version` exist?

**Comment by Meir017** [comment]
@murugaratham looks good, just one question (see review)

File: lib/crawler.js (hunk 1 above)
Comment (murugaratham) [reply lib/crawler.js]: i probably had some funky package.json file at my work place when i tested, i've just tested with my home machine, they shouldn't occur. I've followed up with another commit and removed this line

**Comment by Meir017** [comment]
@murugaratham thanks 👍 just published 3.10.0

**Pull request merged**